sql/information_schema/information_schema.go

```diff
@@ -676,6 +685,107 @@ func triggersRowIter(ctx *Context, c *Catalog) (RowIter, error) {
 	return RowsToRowIter(rows...), nil
 }
 
+func checkConstraintsRowIter(ctx *Context, c *Catalog) (RowIter, error) {
+	var rows []Row
+	for _, db := range c.AllDatabases() {
+		tableNames, err := db.GetTableNames(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tableName := range tableNames {
+			tbl, _, err := c.Table(ctx, db.Name(), tableName)
+			if err != nil {
+				return nil, err
+			}
+
+			checkTbl, ok := tbl.(CheckTable)
+			if ok {
+				checkDefinitions, err := checkTbl.GetChecks(ctx)
+				if err != nil {
+					return nil, err
+				}
+
+				if len(checkDefinitions) == 0 {
+					continue
+				}
+
+				for _, checkDefinition := range checkDefinitions {
+					rows = append(rows, Row{"def", db.Name, checkDefinition.Name, checkDefinition.CheckExpression})
+				}
+			}
+		}
+	}
+
+	return RowsToRowIter(rows...), nil
+}
+
+func tableConstraintRowIter(ctx *Context, c *Catalog) (RowIter, error) {
+	var rows []Row
+	for _, db := range c.AllDatabases() {
+		tableNames, err := db.GetTableNames(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, tableName := range tableNames {
+			tbl, _, err := c.Table(ctx, db.Name(), tableName)
+			if err != nil {
+				return nil, err
+			}
+
+			// Get all the CHECKs
+			checkTbl, ok := tbl.(CheckTable)
+			if ok {
+				checkDefinitions, err := checkTbl.GetChecks(ctx)
+				if err != nil {
+					return nil, err
+				}
+
+				if len(checkDefinitions) == 0 {
+					continue
+				}
+
+				for _, checkDefinition := range checkDefinitions {
+					enforced := "YES"
+					if !checkDefinition.Enforced {
+						enforced = "NO"
+					}
+					rows = append(rows, Row{"def", db.Name, checkDefinition.Name, db.Name(), tbl.Name(), "CHECK", checkDefinition, enforced})
+				}
+			}
+
+			// Get UNIQUEs, PRIMARY KEYs
+			indexTable, ok := tbl.(IndexedTable)
```

Comment:
This doesn't handle primary keys in all cases, because a table can have primary keys but not implement IndexedTable

You'll need to dedupe these

Reply:
Primary key constraints should be named PRIMARY to match MySQL